app/db_models.py

```diff
@@ -345,6 +347,64 @@ def update_obj(self, question, data):
             if field in data:
                 setattr(question, field, data[field])
 
+class FillInTheBlankQuestion(Question):
+    id = db.Column(db.Integer, db.ForeignKey('question.id'), primary_key=True)
+
+    question_text = db.Column(db.String, nullable=False)
+
+    answer = db.Column(db.String, nullable=False)
+
+    __mapper_args__ = {
+        'polymorphic_identity': QuestionType.FILL_IN_THE_BLANK_QUESTION,
+    }
+
+    def get_answer(self):
+        return markdown_to_html(self.answer)
+    
+    def make_question(question_text):
+        answers = {} # dictionary of all the answers and their corresponding text boxes
+        blank = False #determines if the words are the fill in the blank part
+        temp_answer = "" #keeps track of the fill in the blank answer
+
+        new_question = ""
+
+        textbox_number = 1
+
+        for char in question_text:
```

Comment:
Python's string find method will be a much better way to hunt for the starting and stopping substring. For example `my_string.find("^^^")`. [python string.find documentation](https://docs.python.org/3/library/stdtypes.html#str.find)